File: emulinker/src/main/java/org/emulinker/kaillera/model/impl/PlayerTimeoutException.java
package org.emulinker.kaillera.model.impl;

public class PlayerTimeoutException extends Exception {
  private int playerNumber;
  private KailleraUserImpl player;
  private int timeoutNumber = -1;

  public PlayerTimeoutException(int playerNumber, KailleraUserImpl player) {
    this.playerNumber = playerNumber;
    this.player = player;
  }

  public PlayerTimeoutException(int playerNumber, int timeoutNumber) {
    this.playerNumber = playerNumber;
    this.timeoutNumber = timeoutNumber;
  }

  public int getPlayerNumber() {
    return playerNumber;
  }

  public KailleraUserImpl getPlayer() {
    return player;
  }

  public int getTimeoutNumber() {
    return timeoutNumber;
  }

  public void setTimeoutNumber(int timeoutNumber) {
    this.timeoutNumber = timeoutNumber;
  }

  @Override
  public boolean equals(Object o) {
    if (o != null && o instanceof PlayerTimeoutException) {
      PlayerTimeoutException e = (PlayerTimeoutException) o;
      if (e.getPlayerNumber() == getPlayerNumber() && e.getTimeoutNumber() == getTimeoutNumber())
        return true;
    }

    return false;
  }
}

File: emulinker/src/main/java/org/emulinker/kaillera/model/impl/PlayerTimeoutException.kt
package org.emulinker.kaillera.model.impl

import java.lang.Exception

data class PlayerTimeoutException
    constructor(
        val playerNumber: Int, var timeoutNumber: Int = -1, val player: KailleraUserImpl? = null
    ) : Exception() {
  // TODO(nue): Figure out if it's safe to get rid of this equals method.
  override fun equals(o: Any?): Boolean {
    if (o != null && o is PlayerTimeoutException) {
      if (o.playerNumber == playerNumber && o.timeoutNumber == timeoutNumber) return true
    }
    return false
  }
}
